Runtime/RealtimeDatabase/RealtimeDatabase.cs

```diff
@@ -886,7 +886,7 @@ object GenerateBody(object value)
                 string paramString = value.ToString();
                 int length = paramString.Length;
                 //make sure it's not json, if json, leave as it is.
-                if (paramString[0] != '{' && paramString[length - 1] != '}')
+                if (paramString[0] != '{' && paramString[length - 1] != '}' && paramString[length - 1] != '}')
```

Comment:
Same condition is added twice, it will not fix the issue. I believe, we are almost there. Regarding the last review in #7, please update the condition as follows - `paramString.Length <= 1 || (paramString[0] != '{' && paramString[length - 1] != '}')`.